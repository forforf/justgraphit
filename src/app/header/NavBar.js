
import React from 'react';
import {IndexLink, Link} from 'react-router';
import {Tabs, Tab} from 'material-ui/Tabs';

const NavBar = () => (
    <Tabs>
      <Tab label='GraphIt' containerElement={<IndexLink to='/'>Home</IndexLink>} />
      <Tab label='Editor'  containerElement={<Link to='/edit'>Editor</Link>} />
      <Tab label='About'   containerElement={<Link to='/about'>About</Link>} />
    </Tabs>
);

export default NavBar;
