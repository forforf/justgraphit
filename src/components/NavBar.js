import React, {useState} from 'react';
import {Link} from 'react-router-dom';
import {Tab, Tabs} from '@mui/material';


const NavBar = () => {
  const [value, setValue] = useState(0);

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  return (
    <Tabs value={value} onChange={handleChange}>
      <Tab label='GraphIt' to='/' component={Link} />
      <Tab label='Editor'  to='/edit' component={Link} />
      <Tab label='About'   to='/about' component={Link} />
    </Tabs>
  )
};

export default NavBar;
