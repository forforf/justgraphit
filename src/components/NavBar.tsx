import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { Tab, Tabs } from '@mui/material';

const NavBar = (): JSX.Element => {
  const [value, setValue] = useState<number>(0);

  const handleChange = (
    event: React.ChangeEvent<unknown>,
    newValue: number
  ) => {
    setValue(newValue);
  };

  return (
    <Tabs value={value} onChange={handleChange}>
      <Tab label="GraphIt" to="/" component={Link} />
      <Tab label="Editor" to="/edit" component={Link} />
      <Tab label="About" to="/about" component={Link} />
    </Tabs>
  );
};

export default NavBar;
