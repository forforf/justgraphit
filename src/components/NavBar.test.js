import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow'
import {HashRouter} from 'react-router-dom';
import NavBar from './NavBar';


test('deep snapshot', () => {
  const component = renderer.create(
    <HashRouter>
      <NavBar />,
    </HashRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('shallow snapshot', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<NavBar />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot()
});


