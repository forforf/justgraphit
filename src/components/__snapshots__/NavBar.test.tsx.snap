// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deep snapshot 1`] = `
Array [
  <div
    className="MuiTabs-root css-1ujnqem-MuiTabs-root"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": -0,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-orientation={null}
        className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <a
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          href="#/"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
        >
          GraphIt
          <span
            className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
            style={Object {}}
          />
        </a>
        <a
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          href="#/edit"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
        >
          Editor
        </a>
        <a
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          href="#/about"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
        >
          About
        </a>
      </div>
    </div>
  </div>,
  ",",
]
`;

exports[`shallow snapshot 1`] = `
<ForwardRef(Tabs)
  onChange={[Function]}
  value={0}
>
  <ForwardRef(Tab)
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    label="GraphIt"
    to="/"
  />
  <ForwardRef(Tab)
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    label="Editor"
    to="/edit"
  />
  <ForwardRef(Tab)
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    label="About"
    to="/about"
  />
</ForwardRef(Tabs)>
`;
